web/src/test/java/org/hillview/HistogramAccuracyTest.java

```diff
@@ -0,0 +1,123 @@
+    static String privacy_metadata_name = "privacy_metadata.json";
+
+    @Test
+    public void computeAccuracyTest() {
+        PrivacySchema mdSchema = PrivacySchema.loadFromFile(ontime_directory + privacy_metadata_name);
+        Assert.assertNotNull(mdSchema);
+        Assert.assertNotNull(mdSchema.quantization);
+        ColumnQuantization col1 = mdSchema.quantization.get("DepTime");
+        Assert.assertNotNull(col1);
+        Assert.assertTrue(col1 instanceof DoubleColumnQuantization);
+
+        DoubleColumnQuantization dq = (DoubleColumnQuantization)col1;
+
+        // Construct a histogram corresponding to the leaves.
+        // We will manually aggregate buckets as needed for the accuracy test.
+        HistogramRequestInfo info = new HistogramRequestInfo(new ColumnDescription("DepTime", ContentsKind.Double),
+                0, dq.globalMin, dq.globalMax, dq.getIntervalCount());
+        HistogramSketch sk = info.getSketch(col1);
+        DyadicDecomposition dd = info.getDecomposition(col1);
+
+        double epsilon = mdSchema.epsilon(info.cd.name);
+        System.out.println("Epsilon: " + epsilon);
+
+        FileSetDescription fsd = new FileSetDescription();
+        fsd.fileNamePattern = "../data/ontime_private/????_*.csv*";
+        fsd.fileKind = "csv";
+        fsd.schemaFile = "short.schema";
+
+        Empty e = new Empty();
+        LocalDataSet<Empty> local = new LocalDataSet<Empty>(e);
+        String privacyMetadataFile = DPWrapper.privacyMetadataFile(Utilities.getFolder(fsd.fileNamePattern));
+
+        IMap<Empty, List<IFileReference>> finder = new FindFilesMap(fsd);
+        IDataSet<IFileReference> found = local.blockingFlatMap(finder);
+        IMap<IFileReference, ITable> loader = new LoadFilesMap();
+        IDataSet<ITable> table = found.blockingMap(loader);
+        DPWrapper wrapper = new DPWrapper(mdSchema);
+
+        Histogram hist = table.blockingSketch(sk); // Leaf counts.
+        PrivateHistogram ph = new PrivateHistogram(dd, hist, epsilon, false);
+
+        int totalLeaves = dd.getQuantizationIntervalCount();
+        double scale = Math.log(totalLeaves) / Math.log(2);
+        scale /= epsilon;
+        double baseVariance = 2 * Math.pow(scale, 2);
+        LaplaceDistribution dist = new LaplaceDistribution(0, scale); // TODO: (more) secure PRG
+
+        // Do all-intervals accuracy on leaves.
+        int n = 0;
+        double sqtot = 0.0;
+        double abstot = 0.0;
+        for (int left = 0; left < hist.getBucketCount(); left++) {
+            for (int right = left; right < hist.getBucketCount(); right++) {
+                Pair<Double, Double> noise = dd.noiseForRange(left, right, epsilon,
+                        dist, baseVariance, false);
+                sqtot += Math.pow(noise.first, 2);
+                abstot += Math.abs(noise.first);
+                n++;
+            }
+        }
+
+        System.out.println("Bucket count: " + hist.getBucketCount());
```

Comment:
An assert is more useful than a print, but anything is better than nothing.

Reply:
Ah, this is really a benchmark and not a regression test, but I guess accuracy should stay constant here, so I can write asserts instead.